components/core/src/clp_s/SchemaTree.hpp

```diff
@@ -71,7 +88,10 @@
     int32_t m_id;
     int32_t m_parent_id;
     std::vector<int32_t> m_children_ids;
-    std::string m_key_name;
+    // We use a buffer so that references to this key name are stable after this SchemaNode is move
+    // constructed
+    std::unique_ptr<char[]> m_key_buf;
+    std::string_view m_key_name;
```

Comment:
It's mostly automatic because of the unique_ptr, and importantly it allows references to the underlying buffer to remain valid even if the SchemaNode is moved.